src/simtools/production_configuration/extract_mc_event_data.py

```diff
@@ -192,10 +189,58 @@ def _process_mc_event(self, eventio_object, data_lists):
 
     def _process_array_event(self, eventio_object, data_lists):
         """Process array event and update data lists."""
+        tracking_positions = []
+        previous_index = -1
+
         for i, obj in enumerate(eventio_object):
-            if i == 0 and isinstance(obj, TriggerInformation):
+            if isinstance(obj, TriggerInformation):
                 self._process_trigger_information(obj, data_lists)
 
+            if isinstance(obj, TrackingPosition):
+                tracking_position = obj.parse()
+                tracking_positions.append(
+                    {
+                        "altitude": tracking_position["altitude_raw"],
+                        "azimuth": tracking_position["azimuth_raw"],
+                    }
+                )
+
+            if i < previous_index:
+                self._process_tracking_positions(tracking_positions, data_lists)
+                tracking_positions = []  # Reset for the next shower
+
+            previous_index = i
+
+        if tracking_positions:
+            self._process_tracking_positions(tracking_positions, data_lists)
+
+    def _process_tracking_positions(self, tracking_positions, data_lists):
+        """
+        Process the collected tracking positions and update data lists.
+
+        Parameters
+        ----------
+        tracking_positions : list of dict
+            List of tracking positions with "altitude" and "azimuth" keys.
+        data_lists : dict
+            Data lists to update.
+        """
+        altitudes = [pos["altitude"] for pos in tracking_positions]
+        azimuths = [pos["azimuth"] for pos in tracking_positions]
+        if isinstance(altitudes[0], list):
```

Comment:
Can you confirm that the length of the lists are always >0?

Reply:
Thanks! Yes this is the case always.